pkg/network/http/driver_interface.go

```diff
@@ -0,0 +1,238 @@
+	err := d.setupHTTPHandle()
+	if err != nil {
+		return nil, err
+	}
+
+	d.dataChannel = make(chan []driver.HttpTransactionType)
+	return d, nil
+}
+
+func (di *httpDriverInterface) setupHTTPHandle() error {
+	dh, err := driver.NewHandle(windows.FILE_FLAG_OVERLAPPED, driver.HTTPHandle)
+	if err != nil {
+		return err
+	}
+
+	filters, err := createHTTPFilters()
+	if err != nil {
+		return err
+	}
+
+	if err := dh.SetHTTPFilters(filters); err != nil {
+		return err
+	}
+
+	iocp, buffers, err := driver.PrepareCompletionBuffers(dh.Handle, httpReadBufferCount)
+	if err != nil {
+		return err
+	}
+
+	di.driverHTTPHandle = dh
+	di.iocp = iocp
+	di.readBuffers = buffers
+	return nil
+}
+
+func createHTTPFilters() ([]driver.FilterDefinition, error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return nil, err
+	}
+
+	var filters []driver.FilterDefinition
+	for _, iface := range ifaces {
+		// IPv4
+		filters = append(filters, driver.FilterDefinition{
+			FilterVersion:  driver.Signature,
+			Size:           driver.FilterDefinitionSize,
+			Direction:      driver.DirectionOutbound,
+			FilterLayer:    driver.LayerTransport,
+			InterfaceIndex: uint64(iface.Index),
+			Af:             windows.AF_INET,
+			Protocol:       windows.IPPROTO_TCP,
+		}, driver.FilterDefinition{
+			FilterVersion:  driver.Signature,
+			Size:           driver.FilterDefinitionSize,
+			Direction:      driver.DirectionInbound,
+			FilterLayer:    driver.LayerTransport,
+			InterfaceIndex: uint64(iface.Index),
+			Af:             windows.AF_INET,
+			Protocol:       windows.IPPROTO_TCP,
+		})
+
+		// IPv6
+		filters = append(filters, driver.FilterDefinition{
+			FilterVersion:  driver.Signature,
+			Size:           driver.FilterDefinitionSize,
+			Direction:      driver.DirectionOutbound,
+			FilterLayer:    driver.LayerTransport,
+			InterfaceIndex: uint64(iface.Index),
+			Af:             windows.AF_INET6,
+			Protocol:       windows.IPPROTO_TCP,
+		}, driver.FilterDefinition{
+			FilterVersion:  driver.Signature,
+			Size:           driver.FilterDefinitionSize,
+			Direction:      driver.DirectionInbound,
+			FilterLayer:    driver.LayerTransport,
+			InterfaceIndex: uint64(iface.Index),
+			Af:             windows.AF_INET6,
+			Protocol:       windows.IPPROTO_TCP,
+		})
+	}
+
+	return filters, nil
+}
+
+func (di *httpDriverInterface) setMaxFlows(maxFlows uint64) error {
+	log.Debugf("Setting max flows in driver http filter to %v", maxFlows)
+	err := windows.DeviceIoControl(di.driverHTTPHandle.Handle,
+		driver.SetMaxFlowsIOCTL,
+		(*byte)(unsafe.Pointer(&maxFlows)),
+		uint32(unsafe.Sizeof(maxFlows)),
+		nil,
+		uint32(0), nil, nil)
+	if err != nil {
+		log.Warnf("Failed to set max number of flows in driver http filter to %v %v", maxFlows, err)
+	}
+	return err
+}
+
+func (di *httpDriverInterface) startReadingBuffers() {
+	di.eventLoopWG.Add(1)
+	go func() {
+		defer di.eventLoopWG.Done()
+
+		transactionSize := uint32(C.size_of_http_transaction_type())
```

Comment:
Does `unsafe.Sizeof(C.HTTP_TRANSACTION_TYPE)` or `unsafe.Sizeof(HttpTransactionType)` work instead of this custom function?

Reply:
Or `C.sizeof_struct_HttpTransactionType`

Reply:
Unfortunately not. This was the only solution I was able to find which compiles

Reply:
I had a typo, it should be `C.sizeof_struct__HttpTransactionType` (note the double underscore).